src/test/scala/mesosphere/marathon/event/http/HttpEventActorTest.scala

```diff
@@ -0,0 +1,112 @@
+package mesosphere.marathon.event.http
+
+import akka.actor.{ Actor, ActorSystem, Props }
+import akka.testkit.{ EventFilter, TestActorRef }
+import com.codahale.metrics.MetricRegistry
+import com.typesafe.config.ConfigFactory
+import mesosphere.marathon.MarathonSpec
+import mesosphere.marathon.event.EventStreamAttached
+import mesosphere.marathon.event.http.HttpEventActor.EventNotificationLimit
+import mesosphere.marathon.event.http.SubscribersKeeperActor.GetSubscribers
+import mesosphere.marathon.integration.setup.WaitTestSupport.waitUntil
+import mesosphere.marathon.metrics.Metrics
+import mesosphere.util.Mockito
+import org.scalatest.{ GivenWhenThen, Matchers }
+import spray.http.{ HttpRequest, HttpResponse, StatusCode }
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class HttpEventActorTest extends MarathonSpec with Mockito with GivenWhenThen with Matchers {
+
+  test("A message is broadcast to all subscribers") {
+    Given("A HttpEventActor with 2 subscribers")
+    val aut = TestActorRef(new NoHttpEventActor(Set("host1", "host2")))
+
+    When("An event is send to the actor")
+    aut ! EventStreamAttached("remote")
+
+    Then("The message is broadcast to both subscribers")
+    waitUntil("Wait for 2 subscribers to get notified", 1.second) {
+      aut.underlyingActor.requests.size == 2
+    }
+  }
+
+  test("If a message is send to non existing subscribers") {
+    Given("A HttpEventActor with 2 subscribers")
+    val aut = TestActorRef(new NoHttpEventActor(Set("host1", "host2")))
+    responseAction = () => throw new RuntimeException("Can not connect")
+
+    When("An event is send to the actor")
+    aut ! EventStreamAttached("remote")
+
+    Then("The callback listener is rate limited")
+    waitUntil("Wait for rate limiting 2 subscribers", 1.second) {
+      aut.underlyingActor.limiter("host1").deadLine.isDefined && aut.underlyingActor.limiter("host2").deadLine.isDefined
+    }
+  }
+
+  test("If a message is send to a slow subscriber") {
+    Given("A HttpEventActor with 1 subscriber")
+    val aut = TestActorRef(new NoHttpEventActor(Set("host1")))
+    responseAction = () => { Thread.sleep(100); response }
```

Comment:
Ouch, sleeps in tests. Should I less that pass for releasing-0.10? I wouldn't like that on master...

Alternatively, you can virtualize your time by making the method to determine the time `protected` and overridable so that you can mock it for testing.